Sources/RemoteNotification.swift

```diff
@@ -33,28 +33,66 @@ public struct RemoteNotification: CustomStringConvertible, Equatable {
     /// The identifier for the UIUserNotificationCategory associated with the push notification. Derived from userInfo["category"]
     public let categoryIdentifier: String?
     /// Custom fields on the remote notification.
-    public let userInfo: [String : AnyObject]
+    public let userInfo: [String : Any]
     /// The dictionary representation of the remote notification.
-    public let remoteNotificationDictionary: [String : AnyObject]
+    public let remoteNotificationDictionary: [String : Any]
     
-    public init?(remoteNotification: [NSObject : AnyObject]?) {
-        guard let remoteNotification = remoteNotification as? [String : AnyObject] else {
+    // MARK: Equatable
+    
+    public static func ==(lhs: RemoteNotification, rhs: RemoteNotification) -> Bool {
+        if lhs.alert != nil || rhs.alert != nil {
+            guard let lhsAlert = lhs.alert, let rhsAlert = rhs.alert, lhsAlert == rhsAlert else {
```

Comment:
isn't there an == operator on Optional<Equatable>, if not, may be cool to make one?  then you can do 

lhs.alert == rhs. alert, etc. for all these.